File: 4.c
//find the biggest palindrome made from the product of two 3-digit numbers
#include <stdio.h>

int isPalindrome(int n)
{
    int temp = n;
    int rev = 0;
    while (temp != 0)
    {
        rev = rev * 10 + temp % 10;
        temp /= 10;
    }
    if (rev == n)
        return 1;
    else
        return 0;
}

int main()
{
    for (int i = 1000; i >= 100; i--)
    {
        for (int j = 1000; j >= 100; j--)
        {
            if (isPalindrome(i * j))
            {
                printf("%d\n", i * j);
                return 0;
            }
        }
    }

}

File: 4.py
#Problem 4#

#A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.#

#Find the largest palindrome made from the product of two 3-digit numbers.
biggest = 0

def isPalindrome(n):
    return str(n) == str(n)[::-1]

#find the biggest palindrome made from the product of two 3-digit numbers
for i in range(100, 1000):
    for j in range(100, 1000):
        if isPalindrome(i*j) and i*j > biggest:
            biggest = i*j

print(biggest)